### PR TITLE
Drop time when used as a tag or field key

### LINKED ISSUE #1834: Server should disallow tags with name "time"
Those fields have special meaning to the database. Currently if a series has a tag named "id", the output of `show series` has two columns named "id":

```
{
    "results": [
        {
            "series": [
                {
                    "name": "random_measurement_0",
                    "columns": ["id","foo","host","id","region","time"],
                    "values": [
                        [1293,"bar6","server6","id_9","west","time_2"]
                    ]
                }
            ]
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by jsternberg**
The "time" field and tags are unqueryable so we prevent those from being
written so we don't have unreadable data.

Fixes #1834.

### CODE REVIEW

**Comment by gunnaraasen** [comment]
Is this a breaking change? Even though a `time` tag/field cannot be queried, there may be clients expecting to be able to write a `time` tag/field and will break after this change. Seems like dropping the `time` tag/field and adding a warning in the logs or HTTP response would be a more compatible approach.

**Comment by jsternberg** [comment]
That's a good point. I think this would classify as a breaking change.

Comments @pauldix is dropping it ok or should we find a way to make this queryable?

File: tsdb/shard.go [hunk 1]
```diff
@@ -447,6 +447,17 @@ func (s *Shard) validateSeriesAndFields(points []models.Point) ([]*FieldCreate,
 
 	// get the shard mutex for locally defined fields
 	for _, p := range points {
+		// verify the tags and fields
+		tags := p.Tags()
```
Comment (jwilder) [inline tsdb/shard.go]: This isn't the right place to reject this as it will fail the whole batch instead of the point.

Comment (jsternberg) [inline tsdb/shard.go]: We would also have that issue with the max series error. We may want to look into that again if we have the time.

**Comment by pauldix** [comment]
Dropping and adding in the log seems like an ok approach to keep anyone from breaking now. But longer term I think that makes it less usable for new users. If you fail the write then any new user coming in would get immediate feedback that what they're doing isn't supported. Users frequently don't have access to the logs.

I'm ok with the breaking change if we get it before 1.0. Otherwise, drop and warn would have to be it.

**Comment by jsternberg** [comment]
I dropped it and I added a `TODO.md` file to keep track of any issues we can't resolve completely until 2.0. This way, we can remember which things we would want to break when we get to that point and decide then if we want to do it or not.

**Comment by gunnaraasen** [comment]
👍 the 2.0 TODO will be useful. Although it feels like it should be a GH issue of wiki page so updating it doesn't require a PR.

**Comment by jsternberg** [comment]
@gunnaraasen my idea was that people would update it whenever we made a change like this where we worked around something. I'll merge this and we can always remove it later if we decide that a wiki would be better.

**Pull request merged**